File: DependencyInjection/DependencyInjectionAfter/_1_ControlFreakRefactoredToDependencyInjection/Outbound/XmlOutboundMessage.cs
using DependencyInjectionAfter._1_ControlFreakRefactoredToDependencyInjection.Interfaces;

namespace DependencyInjectionAfter._1_ControlFreakRefactoredToDependencyInjection.Outbound
{
  public interface IOutboundMessage
  {
    void SendVia(ISocket outputSocket);
  }

  public class XmlOutboundMessage : DataDestination, IOutboundMessage
  {
    private readonly XmlMarshalling _xmlMarshalling;
    private string _content = string.Empty;

    public XmlOutboundMessage(XmlMarshalling xmlMarshalling)
    {
      _xmlMarshalling = xmlMarshalling;
    }

    public void SendVia(ISocket outputSocket)
    {
      var marshalledContent = _xmlMarshalling.Of(_content);
      outputSocket.Open();
      outputSocket.Send(marshalledContent);
      outputSocket.Close();
    }

    public void Add(string s)
    {
      _content += s;
    }
  }
}

File: Java/dependency-injection/control-freak-refactored/src/main/java/com/github/grzesiek_galezowski/DependencyInjectionAfter/Outbound/XmlOutboundMessage.java
package com.github.grzesiek_galezowski.DependencyInjectionAfter.Outbound;

import com.github.grzesiek_galezowski.DependencyInjectionAfter.Interfaces.DataDestination;

public class XmlOutboundMessage implements DataDestination, IOutboundMessage {
  private final XmlMarshalling _xmlMarshalling;
  private String _content = "";

  public XmlOutboundMessage(XmlMarshalling xmlMarshalling) {
    _xmlMarshalling = xmlMarshalling;
  }

  public void SendVia(ISocket outputSocket) {
    String marshalledContent = _xmlMarshalling.Of(_content);
    outputSocket.Open();
    outputSocket.Send(marshalledContent);
    outputSocket.Close();
  }

  public void Add(String s) {
    _content += s;
  }
}
